test/test-cli.py

```diff
@@ -326,6 +327,26 @@ def test_language_typescript_warns_about_undeclared_variables_despite_eval(
         self.assertEqual(result.stdout, "")
         self.assertEqual(result.returncode, 1)
 
+    def test_auto_hyperlinks_are_not_printed_in_dumb_terminals(self) -> None:
+        with tempfile.TemporaryDirectory() as test_directory:
+            test_file = pathlib.Path(test_directory) / "test.js"
+            test_file.write_text("var parenthesesMissing;\nif parenthesesMissing { }\n")
+
+            (pid, fd) = pty.fork()
+            if pid == 0:
+                qljs = get_quick_lint_js_executable_path()
+                os.execve(qljs, [qljs, str(test_file)], {"TERM": "dumb"})
+            else:
+                (_, rc) = os.waitpid(pid, 0)
+                stderr = os.read(fd, 4096)
+                returncode = 0
+                if os.WIFEXITED(rc):
+                    returncode = os.WEXITSTATUS(rc)
+                self.assertEqual(returncode, 1)
+                self.assertNotIn(
+                    b"[\x1b]8;;https://quick-lint-js.com/errors/E0017/\x1b\E0017\x1b]8;;\x1b\]",
```

Comment:
I think it's safer to assert that just `\x1b` is not present. We might change the exact formatting or the URL, and doing so would cause this test to always pass.